tests/functional/object/mcg/test_sts_client.py

```diff
@@ -0,0 +1,264 @@
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.fixture()
+def nb_assign_user_role_fixture(request, mcg_obj_session):
+
+    email = None
+
+    def factory(user_email, role_name, principal="*"):
+        """
+        Assign assume role policy to the user
+
+        Args:
+            user_email (str): Name/id/email of the user
+            role_name (str): Name of the role
+
+        """
+        nonlocal email
+        email = user_email
+        noobaa_assume_role_policy = (
+            f'{{"role_name": "{role_name}","assume_role_policy": '
+            f'{{"version": "2024-07-16","statement": [{{"action": ["sts:AssumeRole"],'
+            f'"effect": "allow","principal": ["{principal}"]}}]}}}}'
+        )
+
+        mcg_obj_session.assign_sts_role(user_email, noobaa_assume_role_policy)
+
+    def teardown():
+        """
+        Remove role from the user
+
+        """
+        try:
+            mcg_obj_session.remove_sts_role(email)
+        except CommandFailed as e:
+            if "No such account email" not in e.args[0]:
+                raise
+
+    request.addfinalizer(teardown)
+    return factory
+
+
+@pytest.fixture()
+def nb_account_factory(request):
+    """
+    Fixture to create noobaa account
+
+    """
+    nb_account = []
+
+    def factory(mcg, user_name, user_email, **kwargs):
+        """
+        Factory function to create the noobaa account
+
+        Args:
+            mcg (MCG): MCG object
+            user_name (str): User name
+            user_email (str): Email of the user
+
+        """
+        account = NoobaaAccount(
+            mcg=mcg, name=f"{user_name}", email=f"{user_email}", **kwargs
+        )
+        nb_account.append(account)
+
+        return account
+
+    def teardown():
+        """
+        Teardown function that deletes the Noobaa account
+
+        """
+        for account in nb_account:
+            account.delete_account()
+
+    request.addfinalizer(teardown)
+    return factory
+
+
+@pytest.fixture()
+def new_bucket(request, mcg_obj_session):
+    """
+    Create new bucket using s3
+
+    """
+    buckets_created = []
+
+    def factory(bucket_name, s3_client=None):
+        s3_create_bucket(mcg_obj_session, bucket_name, s3_client)
+        logger.info(f"Created new-bucket {bucket_name}")
+        bucket_obj = {
+            "s3client": s3_client,
+            "bucket": bucket_name,
+            "mcg": mcg_obj_session,
+        }
+        buckets_created.append(bucket_obj)
+
+    def finalizer():
+        """
+        Cleanup the created bucket
+
+        """
+        for bucket in buckets_created:
+            logger.info(f"Deleting bucket {bucket.get('bucket')}")
+            s3_delete_bucket(
+                bucket.get("mcg"), bucket.get("bucket"), bucket.get("s3client")
+            )
+            logger.info(f"Deleted the bucket {bucket}")
+
+    request.addfinalizer(finalizer)
+    return factory
+
+
```

Comment:
can this be run on any platforms or it is specific to only aws?

Reply:
can run on any platform. Not platform specific.